test/jdk/java/time/test/java/time/format/TestDateTimeFormatterBuilder.java

```diff
@@ -526,6 +530,186 @@ public void test_appendZoneText_1arg_nullText() throws Exception {
+    }
+
+    @Test (dataProvider="dayPeriodParsePattern")
+    public void test_dayPeriodParsePattern(String pattern, String hourDayPeriod, long expected) throws Exception {
+        builder.appendPattern(pattern);
+        DateTimeFormatter f = builder.toFormatter().withLocale(Locale.US);
+        var p = f.parse(hourDayPeriod);
+        assertEquals(p.getLong(HOUR_OF_DAY), expected);
+    }
+
+    @DataProvider(name="dayPeriodParseInvalid")
+    Object[][] data_dayPeriodParseInvalid() {
+        return new Object[][] {
+                {TextStyle.FULL, Locale.US, "00:01 midnight"},
+                {TextStyle.FULL, Locale.US, "06:01 at night"},
+                {TextStyle.FULL, Locale.US, "05:59 in the morning"},
+                {TextStyle.FULL, Locale.US, "11:59 noon"},
+                {TextStyle.FULL, Locale.US, "18:00 in the afternoon"},
+                {TextStyle.FULL, Locale.US, "17:59 in the evening"},
+                {TextStyle.NARROW, Locale.US, "00:01 mi"},
+                {TextStyle.NARROW, Locale.US, "06:01 at night"},
+                {TextStyle.NARROW, Locale.US, "05:59 in the morning"},
+                {TextStyle.NARROW, Locale.US, "11:59 n"},
+                {TextStyle.NARROW, Locale.US, "18:00 in the afternoon"},
+                {TextStyle.NARROW, Locale.US, "17:59 in the evening"},
+
+                {TextStyle.FULL, Locale.JAPAN, "00:01 \u771f\u591c\u4e2d"},
+                {TextStyle.FULL, Locale.JAPAN, "04:00 \u591c\u4e2d"},
+                {TextStyle.FULL, Locale.JAPAN, "03:59 \u671d"},
+                {TextStyle.FULL, Locale.JAPAN, "12:01 \u6b63\u5348"},
+                {TextStyle.FULL, Locale.JAPAN, "16:00 \u663c"},
+                {TextStyle.FULL, Locale.JAPAN, "19:01 \u5915\u65b9"},
+                {TextStyle.FULL, Locale.JAPAN, "23:00 \u591c"},
+                {TextStyle.NARROW, Locale.JAPAN, "00:01 \u771f\u591c\u4e2d"},
+                {TextStyle.NARROW, Locale.JAPAN, "04:00 \u591c\u4e2d"},
+                {TextStyle.NARROW, Locale.JAPAN, "03:59 \u671d"},
+                {TextStyle.NARROW, Locale.JAPAN, "12:01 \u6b63\u5348"},
+                {TextStyle.NARROW, Locale.JAPAN, "16:00 \u663c"},
+                {TextStyle.NARROW, Locale.JAPAN, "19:01 \u5915\u65b9"},
+                {TextStyle.NARROW, Locale.JAPAN, "23:00 \u591c"},
+        };
+    }
+    @Test (dataProvider="dayPeriodParseInvalid")
+    public void test_dayPeriodParseInvalid(TextStyle ts, Locale l, String dayPeriod) throws Exception {
+        try {
+            builder.append(ISO_LOCAL_TIME).appendLiteral(' ').appendDayPeriodText(ts)
+                    .toFormatter()
+                    .withLocale(l)
+                    .parse(dayPeriod);
+            throw new RuntimeException("DateTimeParseException should be thrown");
+        } catch (DateTimeParseException e) {
+            assertEquals(e.getCause().getMessage(),
+                    "Conflict found: " + HOUR_OF_DAY + " conflict with day period");
+        }
+    }
+
```

Comment:
There don't seem to be any tests of the resolver logic in `Parsed`.

Reply:
Test case added.